packages/core/lib/src/internal/flattener.dart

```diff
@@ -8,24 +8,24 @@ List<Flattened> flatten(BuildTree tree) => _instance.flatten(tree);
 
 @immutable
 class Flattened {
-  final SpanBuilder spanBuilder;
-  final Widget widget;
-  final WidgetBuilder widgetBuilder;
+  final SpanBuilder? spanBuilder;
+  final Widget? widget;
+  final WidgetBuilder? widgetBuilder;
 
   Flattened._({this.spanBuilder, this.widget, this.widgetBuilder});
 }
 
-typedef SpanBuilder = InlineSpan Function(BuildContext);
+typedef SpanBuilder = InlineSpan? Function(BuildContext);
 
 final _instance = _Flattener();
 
 class _Flattener {
-  List<Flattened> _flattened;
-  StringBuffer _buffer, _prevBuffer;
-  _Recognizer _recognizer, _prevRecognizer;
-  List<dynamic> _spans;
-  bool _swallowWhitespace;
-  TextStyleBuilder _tsb, _prevTsb;
+  late List<Flattened> _flattened;
```

Comment:
I don't think I understand... `late` here just means that we promise that `_flattened` is initialized when we first use it. I think we could make even more fields in this class `late`, because `_resetLoop` initializes them all bevore they are used.